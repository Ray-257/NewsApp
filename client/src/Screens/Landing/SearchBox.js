import React from "react";

const SearchBox = ({ handleSearch,searchKey,setQuery }) => {
  return (
    <div className="search">
      <form onSubmit={handleSearch}>
        <input
          autoFocus
          value={searchKey}
          onChange={(e) => setQuery(e.target.value)}
        />
        <button>Search</button>
      </form> 
    </div>
  );
};


export default SearchBox;
