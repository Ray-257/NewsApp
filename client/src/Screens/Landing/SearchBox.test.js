import React from 'react';
import { shallow } from 'enzyme';
import SearchBox from './SearchBox';


describe('Search Box', () => {
  let wrapper
  const handleSearch = jest.fn();
  const setQuery = jest.fn();
  beforeEach(() => {
     wrapper = shallow(<SearchBox searchKey="test" handleSearch={handleSearch} setQuery={setQuery}  />);
  })
  afterEach(() => {
    wrapper = null
  })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have input value as `test', () => {
    const inputValue = wrapper.find('input').get(0).props.value;
    expect(inputValue).toEqual("test");
  });
  it('should simulate event on search input', () => {
    wrapper.find('input').simulate('change',{target:''})
    expect(setQuery).toHaveBeenCalledTimes(1);
  });
  it('should simulate event on form  submit', () => {
    wrapper.find('form').simulate('click')
    expect(setQuery).toHaveBeenCalledTimes(1);
  });
});