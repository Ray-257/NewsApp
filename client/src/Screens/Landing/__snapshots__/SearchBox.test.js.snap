// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box should render correctly 1`] = `ShallowWrapper {}`;
